internal/controller/nonadminbackup_controller_test.go

```diff
@@ -465,69 +491,94 @@ var _ = ginkgo.Describe("Test full reconcile loop of NonAdminBackup Controller",
+				select {
+				case <-ctx.Done():
+					return false, ctx.Err()
+				default:
+					// Check if the manager has started by verifying if the client is initialized
+					return k8sManager.GetClient() != nil, nil
+				}
+			})
+			// Check if the context timeout or another error occurred
+			gomega.Expect(err).ToNot(gomega.HaveOccurred(), "Manager failed to start within the timeout period")
 
 			ginkgo.By("Waiting Reconcile of create event")
-			nonAdminBackup := buildTestNonAdminBackup(nonAdminNamespaceName, scenario.spec)
-			gomega.Expect(k8sClient.Create(ctx, nonAdminBackup)).To(gomega.Succeed())
+			nonAdminBackup := buildTestNonAdminBackup(nonAdminObjectNamespace, nonAdminObjectName, scenario.spec)
+			gomega.Expect(k8sClient.Create(ctxTimeout, nonAdminBackup)).To(gomega.Succeed())
 			// wait NAB reconcile
-			time.Sleep(1 * time.Second)
+			time.Sleep(2 * time.Second)
 
 			ginkgo.By("Fetching NonAdminBackup after Reconcile")
 			gomega.Expect(k8sClient.Get(
-				ctx,
+				ctxTimeout,
 				types.NamespacedName{
-					Name:      testNonAdminBackupName,
-					Namespace: nonAdminNamespaceName,
+					Name:      nonAdminObjectName,
+					Namespace: nonAdminObjectNamespace,
 				},
 				nonAdminBackup,
 			)).To(gomega.Succeed())
 
 			ginkgo.By("Validating NonAdminBackup Status")
-			gomega.Expect(checkTestNonAdminBackupStatus(nonAdminBackup, scenario.status, nonAdminNamespaceName, oadpNamespaceName)).To(gomega.Succeed())
 
-			if scenario.status.VeleroBackup != nil && len(scenario.status.VeleroBackup.Name) > 0 {
+			gomega.Expect(checkTestNonAdminBackupStatus(nonAdminBackup, scenario.status, oadpNamespace)).To(gomega.Succeed())
+
+			if scenario.status.VeleroBackup != nil && len(nonAdminBackup.Status.VeleroBackup.NameUUID) > 0 {
 				ginkgo.By("Checking if NonAdminBackup Spec was not changed")
 				gomega.Expect(reflect.DeepEqual(
 					nonAdminBackup.Spec,
 					scenario.spec,
 				)).To(gomega.BeTrue())
 
 				ginkgo.By("Simulating VeleroBackup update to finished state")
+
 				veleroBackup := &velerov1.Backup{}
 				gomega.Expect(k8sClient.Get(
-					ctx,
+					ctxTimeout,
 					types.NamespacedName{
-						Name:      function.GenerateVeleroBackupName(nonAdminNamespaceName, testNonAdminBackupName),
-						Namespace: oadpNamespaceName,
+						Name:      nonAdminBackup.Status.VeleroBackup.NameUUID,
+						Namespace: oadpNamespace,
 					},
 					veleroBackup,
 				)).To(gomega.Succeed())
-				veleroBackup.Status.Phase = velerov1.BackupPhaseCompleted
-				// TODO can not call .Status().Update() for veleroBackup object: backups.velero.io "name..." not found error
-				gomega.Expect(k8sClient.Update(ctx, veleroBackup)).To(gomega.Succeed())
+
+				veleroBackup.Status = velerov1.BackupStatus{
+					Phase: velerov1.BackupPhaseCompleted,
+				}
+
+				gomega.Expect(k8sClient.Update(ctxTimeout, veleroBackup)).To(gomega.Succeed())
+
+				time.Sleep(1 * time.Second)
```

Comment:
this is unnecessary, right?

Reply:
I think it was necessary by the time of writing as I think I was getting some error. Can't remember now.

Reply:
Removed - ran the tests multiple times and it's not required. Thanks.